src/fides/api/util/tcf/tcf_experience_contents.py

```diff
@@ -388,6 +391,7 @@ def build_purpose_or_feature_section_and_update_vendor_map(
                 non_vendor_record_map=non_vendor_record_map,
                 is_purpose_type=is_purpose_section,
                 use_or_feature=attribute,
+                declaration=privacy_declaration_row,
```

Comment:
An edge case - say a system has two data uses, both of which map to the same purpose, but with different retention periods. For example, a system has marketing.advertising.first_party.contextual with a retention period of 100 days and a separate declaration marketing.advertising.frequency_capping with a retention period of 90 days. Both map to Purpose 2. Only one retention period will be kept (I am ordering the previous query, so at least the retention period kept is repeatable).

Reply:
i think this edge case that's only applicable to non-GVL vendors. which i know is a use case we're handling, so this is good to call out - but i think we can live with some "strange" behavior in the overlay (whatever we think it should be -- perhaps the higher value retention period should be chosen?).

this shouldn't be applicable to GVL vendors, since all of the data use declarations that are generated from/based on a common purpose should, necessarily, have the same retention period. we don't enforce that in fides at all, but in the conversion of GVL --> compass records, that's the logic.

@pattisdr @allisonking let me know if i'm misunderstanding, and also curious to hear what you think on the suggestion to use the higher value retention period on this edge case. thanks for calling it out @pattisdr!

Reply:
mm yeah you are both making sense to me :) I agree using the higher retention period makes sense, though I'm not sure how to do that... would I have to change [this order by](https://github.com/ethyca/fides/blob/aking/4255/tcf-fields/src/fides/api/util/tcf/tcf_experience_contents.py/#L212-L214) to be by retention period instead? or maybe I sort it here? https://github.com/ethyca/fides/blob/aking/4255/tcf-fields/src/fides/api/util/tcf/tcf_experience_contents.py#L390

Reply:
interesting idea that first idea should work fine

Reply:
wait a minute, that's not going to work well if retention period is a string right...

Reply:
if we think that change is too impactful/risky/cumbersome, i'm also not opposed to us just choosing a retention period arbitrarily, since this is such an edge case at the moment...

whatever we choose, we should probably just communicate the behavior to product, since we're kinda in uncharted territory here and making some (reasonable) assumptions/decisions!

Reply:
ah right Allison! In that case, I'd leave as-is.   This edge case is very unlikely, I only added it as I was trying to make sure I'd thought through all the scenarios.  The important thing you've already addressed with moving to adding retention at the embedded purpose level